lib/simulator-xcode-11.4.js

```diff
@@ -0,0 +1,189 @@
+    super(udid, xcodeVersion);
+  }
+
+  /**
+   * @override
+   * Sets UI appearance style.
+   * This function can only be called on a booted simulator.
+   *
+   * @since Xcode SDK 11.4
+   * @param {string} value one of possible appearance values:
+   * - dark: to switch to the Dark mode
+   * - light: to switch to the Light mode
+   */
+  async setAppearance (value) {
+    await this.simctl.setAppearance(_.toLower(value));
+  }
+
+  /**
+   * @override
+   * Gets the current UI appearance style
+   * This function can only be called on a booted simulator.
+   *
+   * @since Xcode SDK 11.4
+   * @returns {string} the current UI appearance style.
+   * Possible values are:
+   * - dark: to switch to the Dark mode
+   * - light: to switch to the Light mode
+   */
+  async getAppearance () {
+    return await this.simctl.getAppearance();
+  }
+
+  /**
+   * @typedef {Object} CertificateOptions
+   * @property {boolean} isRoot [true] - Whether to install the given
+   * certificate into the Trusted Root store (`true`) or to the keychain
+   * (`false`)
+   */
+
+  /**
+   * @override
+   * Adds the given certificate to the booted simulator.
+   * The simulator could be in both running and shutdown states
+   * in order for this method to run as expected.
+   *
+   * @param {string} payload the content of the PEM certificate
+   * @param {CertificateOptions} opts
+   */
+  async addCertificate (payload, opts = {}) {
+    const {
+      isRoot = true,
+    } = opts;
+    const tmpRoot = await tempDir.openDir();
+    try {
+      const certPath = path.resolve(tmpRoot, 'certificate.pem');
+      log.info(`Installing SSL ${isRoot ? 'root ' : ''}certificate '${_.truncate(payload, {length: 20})}'`);
+      if (isRoot) {
+        await this.simctl.addRootCertificate(certPath);
+      } else {
+        await this.simctl.addCertificate(certPath);
+      }
+    } finally {
+      await fs.rimraf(tmpRoot);
+    }
+  }
+
+  /**
+   * @override
+   * Simulates push notification delivery to the booted simulator
+   *
+   * @since Xcode SDK 11.4
+   * @param {Object} payload - The object that describes Apple push notification content.
+   * It must contain a top-level "Simulator Target Bundle" key with a string value matching
+   * the target application‘s bundle identifier and "aps" key with valid Apple Push Notification values.
+   * For example:
+   * {
+   *   "Simulator Target Bundle": "com.apple.Preferences",
+   *   "aps": {
+   *     "alert": "This is a simulated notification!",
+   *     "badge": 3,
+   *     "sound": "default"
+   *   }
+   * }
+   */
+  async pushNotification (payload) {
+    await this.simctl.pushNotification(payload);
+  }
+
+  /**
+   * @override
+   * Sets the permissions for the particular application bundle.
+   *
+   * @param {string} bundleId - Application bundle identifier.
+   * @param {Object} permissionsMapping - A mapping where kays
+   * are service names and values are their corresponding status values.
```

Comment:
Might be nice to know what the possible status values are.

Reply:
np, I'll add more documentation

Reply:
Ahm, I think we already have the list there